autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -51,14 +53,25 @@ public ScalingActivity create(Cluster cluster, ActivityStatus activityStatus, St
         return scalingActivityRepository.save(scalingActivity);
     }
 
-    public void updateWithFlowIdAndTriggerStatus(FlowIdentifier flowIdentifier, String activityCrn, ActivityStatus activityStatus) {
-        ScalingActivity scalingActivity = findByCrn(activityCrn);
+    public void update(Long activityId, FlowIdentifier flowIdentifier, ActivityStatus activityStatus, String reason) {
+        ScalingActivity scalingActivity = scalingActivityRepository.findById(activityId).orElseThrow(notFound("ScalingActivity", activityId));
         scalingActivity.setFlowId(flowIdentifier.getPollableId());
         scalingActivity.setActivityStatus(activityStatus);
-        LOGGER.info("Updating ScalingActivity: {} with FlowInformation and ActivityStatus: {}", activityCrn, activityStatus);
+        LOGGER.info("Updating ScalingActivity: {} with FlowInformation and ActivityStatus: {}", scalingActivity.getActivityCrn(), activityStatus);
+        scalingActivity.setScalingActivityReason(reason);
         save(scalingActivity);
     }
 
+    public void setActivityStatusForIds(Collection<Long> ids, ActivityStatus status) {
+        scalingActivityRepository.setActivityStatusesInIds(ids, status);
```

Comment:
See notes elsewhere about this missing a save call.

Reply:
Since this is a `@Modifying` repository call, `save` need not be invoked. I've followed the pattern in the code to modify records in this manner

Reply:
Sounds good.